super_calendar/super_calendar.py

```diff
@@ -128,7 +128,9 @@ def _generate_record_from_line(self, cr, uid, configurator, line, context):
                     'res_id': line.name.model + ',' + str(record['id']),
                     'model_id': line.name.id,
                 }
-                return super_calendar_values
+                super_calendar_pool.create(cr, uid, super_calendar_values,
```

Comment:
Don't make the create call in the same method. Return the dictionary and make the create where it is (although you remove the too early return. This is to easy inheritance if someone wants to extend the method adding values to the dictionary. It is even mode desirable to make a method that works for only one line and returns a dictionary.